hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/defaultserializers/JavaDefaultSerializers.java

```diff
@@ -470,21 +476,24 @@ public void write(ObjectDataOutput out, HazelcastJsonValue object) throws IOExce
         public HazelcastJsonValue read(ObjectDataInput in) throws IOException {
             return new HazelcastJsonValue(in.readString());
         }
+    }
+
+    public static final class ByteBufferSerializer implements ByteArraySerializer<ByteBuffer> {
 
         @Override
         public int getTypeId() {
-            return JAVASCRIPT_JSON_SERIALIZATION_TYPE;
+            return JAVA_DEFAULT_BYTE_BUFFER;
         }
-    }
-
-    private abstract static class SingletonSerializer<T> implements StreamSerializer<T> {
 
         @Override
-        public void destroy() {
+        public byte[] write(ByteBuffer buffer) throws IOException {
+            return buffer.array();
```

Comment:
`array` is not always available. If we do not support such cases (might be fine for Avro but someone else might also use this serializer), add a comment

Reply:
287b8557